packages/truffle-migrate/migration.js

```diff
@@ -94,9 +94,20 @@ class Migration {
       const Migrations = resolver.require("./Migrations.sol");
 
       if (Migrations && Migrations.isDeployed()) {
-        await self.emitter.emit('saveMigration', self.number);
+        const message = `Saving migration to chain.`;
+
+        if (!this.dryRun){
+          const data = { message: message }
+          await self.emitter.emit('startTransaction', data);
+        }
```

Comment:
Why not during dry-run?

Reply:
Hmmm maybe this should be removed. The dry-run has had abbreviated output and was just preserving here but now this logic is outside the reporter so maybe shouldn't do that. 

On the other hand it's all disposable, like the dry-run doesn't really pick up from a migration that left off or write artifacts that get preserved etc.

Reply:
Hm, then I think this is fine for now. I think eventually we will want a "dry run reporter" that is separate from the regular reporter. That way we can get rid of this conditional logic.

Reply:
Ah yes nice! That's a cool idea.